framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -464,8 +472,9 @@ void MediaRecorderImpl::setRecorderDataSource(std::shared_ptr<stream::OutputData
 		return notifySync();
 	}
 
-	mOutputDataSource = dataSource;
-	mOutputDataSource->setRecorder(shared_from_this());
+	dataSource->setRecorder(shared_from_this());
+	mOutputHandler.setRecorder(shared_from_this());
+	mOutputHandler.setOutputDataSource(dataSource);
```

Comment:
I think this line shows high cohesion among 'DataSource', 'OutputHandler', and 'Recorder'.
Isn't there another way to cut dependencies among those classes...?

Reply:
If we want cut dependencies between datasource and recorder, bufferdatasource logic should be changed.
It is out of scope in this pr

Reply:
Right.
Let's discuss and optimize our architecture later.
And let me modify some expression in my previous comment, i.e., "cut -> reduce".